Dockerfile

```diff
@@ -1,66 +1,54 @@
 #syntax=docker/dockerfile:1.2
 
-FROM golang:1.18 as builder
+FROM golang:1.18-alpine3.16 as builder
 
-RUN apt-get update && apt-get --no-install-recommends install -y \
+RUN apk update && apk add --no-cache \
     git \
     make \
-    apt-utils \
-    apt-transport-https \
     ca-certificates \
     wget \
-    gcc && \
-    apt-get clean \
-    && rm -rf \
-        /var/lib/apt/lists/* \
-        /tmp/* \
-        /var/tmp/* \
-        /usr/share/man \
-        /usr/share/doc \
-        /usr/share/doc-base
-
-WORKDIR /tmp
-
-# https://blog.container-solutions.com/faster-builds-in-docker-with-go-1-11
+    curl \
+    gcc \
+    bash \
+    jq \
+    mailcap
```

Comment:
We might need to double-check whether all these changes are okay to avoid causing any missing dependencies/binaries.

Reply:
We do use the distroless static base image for the final stages in the Dockerfile: https://github.com/argoproj/argo-workflows/pull/10135/files/9987392605dff9421a1beb8c8d8e562979bd3724#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R85

That means none of the binaries from the builder stage get included unless explicitly copied from the base layer. Also, the final steps will complain if there's something missing from the builder step. I had to add `mailcap` to get the `/etc/mime.types` to show up